### DISCUSSION TITLE
Any plan to expose swap chain size configuration?

### DISCUSSION THREAD

**Discussion by ZhennanWu** [gfx-rs/wgpu#3223] [General]
Currently the swap chain size already exists in `wgpu_hal::SurfaceConfiguration::swap_chain_size` with the default size of `wgpu_core::present::DESIRED_NUM_FRAMES = 3`. Any particular reason to block it from the users? Since in some circumstances double buffering is more preferrable than triple buffering. 

And in any case, the current configuration fails to control the actual swap chain size on Vulkan platforms anyway (since Vulkan spec only mandates the minimum image count).

**Comment by cwfitzgerald** [comment]
Yeah we have some, see https://github.com/gfx-rs/wgpu/issues/2869's "Presentation Wait". The tricky part is that the length of the swapchain doesn't really help a whole lot in terms of latency and pacing.

> And in any case, the current configuration fails to control the actual swap chain size on Vulkan platforms anyway (since Vulkan spec only mandates the minimum image count).

How so, it controls it as much as it can. There's a minimum and maximum size given by the adapter, and we clamp our value to that range.
